sway-core/src/semantic_analysis/ast_node/declaration/trait.rs

```diff
@@ -85,20 +82,26 @@ impl ty::TyTraitDeclaration {
                 suffix: name.clone(),
                 is_absolute: false,
             },
-            insert_type(TypeInfo::SelfType),
+            self_type,
             trait_fns
                 .iter()
                 .map(|x| x.to_dummy_func(Mode::NonAbi))
                 .collect(),
         );
+
         // check the methods for errors but throw them away and use vanilla [FunctionDeclaration]s
-        let ctx = ctx.with_self_type(insert_type(TypeInfo::SelfType));
-        let _methods = check!(
-            type_check_trait_methods(ctx, methods.clone()),
-            vec![],
-            warnings,
-            errors
-        );
+        let _methods = methods
+            .iter()
+            .map(|method| {
+                check!(
```

Comment:
If `_methods` isn't used again and all we're doing here is the checks, then it might as well be a for-loop and not `collect()` into a new (unused) `Vec`.